ui/src/pages/publications/[id]/index.tsx

```diff
@@ -200,19 +206,32 @@ const Publication: Types.NextPage<Props> = (props): React.ReactElement => {
     );
 
     const onBookmarkHandler = async () => {
-        if (isBookmarked) {
+        if (showBookmark) {
             //delete the bookmark
             try {
-                await api.destroy(`publications/${publicationData?.id}/bookmark`, user?.token);
-                setIsBookmarked(false);
+                await api.destroy(`bookmarks/${bookmarkId}`, user?.token);
+                setShowBookmark(false);
```

Comment:
```suggestion
                setBookmarkId(null);
```

We can replace with 'setBookmarkId'